ui/src/app/applications/components/applications-sync-panel/applications-sync-panel.tsx

```diff
@@ -38,16 +40,40 @@ export const ApplicationsSyncPanel = ({show, apps, hide}: {show: boolean; apps:
                     <Form
                         defaultValues={{syncFlags: []}}
                         onSubmit={async (params: any) => {
+                            setPending(true);
                             const selectedApps = getSelectedApps(params);
+                            const syncFlags = {...params.syncFlags} as SyncFlags;
+                            const force = syncFlags.Force || false;
+                            if (force) {
+                                const confirmed = await ctx.popup.confirm('Synchronize with force?', () => (
+                                    <div>
+                                        <i className='fa fa-exclamation-triangle' style={{color: ARGO_WARNING_COLOR}} /> {FORCE_WARNING} Are you sure you want to continue?
+                                    </div>
+                                ));
+                                if (!confirmed) {
+                                    setPending(false);
+                                    return;
+                                }
+                            }
```

Comment:
How about we abstract this snippet to a util method? So that we don't need to repeat it.

Reply:
Thank you for the suggestion. This one is a bit hard to abstract and seems overkilling since:
1. We need to modify the state, e.g. `setPending(false)`
2. The logic is specific to applications sync so it's not very suitable as utils.